agent/provider/src/payments/payments.rs

```diff
@@ -494,82 +501,106 @@ impl Handler<UpdateCost> for Payments {
     type Result = ActorResponse<Self, (), Error>;
 
     fn handle(&mut self, msg: UpdateCost, _ctx: &mut Context<Self>) -> Self::Result {
-        let agreement = match self.agreements.get(&msg.invoice_info.agreement_id) {
-            Some(agreement) => agreement,
-            None => {
-                let err_msg = format!(
-                    "Not my activity - agreement [{}].",
-                    &msg.invoice_info.agreement_id
-                );
-                log::warn!("{}", &err_msg);
-
-                return ActorResponse::reply(Err(anyhow!(err_msg)));
-            }
+        let agreement = match self
+            .agreements
+            .get(&msg.invoice_info.agreement_id)
+            .ok_or(anyhow!(
+                "Not my activity - agreement [{}].",
+                &msg.invoice_info.agreement_id
+            ))
+            .log_warn_msg("[UpdateCost]")
+        {
+            Ok(agreement) => agreement,
+            Err(e) => return ActorResponse::reply(Err(e)),
         };
 
-        if let Some(activity) = agreement.activities.get(&msg.invoice_info.activity_id) {
-            if let ActivityPayment::Running { .. } = activity {
+        return match agreement.activities.get(&msg.invoice_info.activity_id) {
+            Some(ActivityPayment::Running { .. }) => {
                 let payment_model = agreement.payment_model.clone();
                 let context = self.context.clone();
+
+                let update_interval = agreement.update_interval;
+                let last_debit_note = agreement.last_send_debit_note.clone();
+                let deadline = agreement.payment_deadline.clone();
```

Comment:
This is not a `deadline` (absolute point in time), but a `timeout` (relative -- duration). Name misled me to understand code.